helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -90,7 +117,24 @@ public synchronized boolean deleteMetadataStoreRealm(String realm) {
       return _zkClient.delete(MetadataStoreRoutingConstants.ROUTING_DATA_PATH + "/" + realm);
     }
 
-    // TODO: Forward the request to leader
+    String leaderHostName = _leaderElection.getCurrentLeaderInfo().getId();
+    String url =
+        leaderHostName + "/admin/v2/namespaces/" + _namespace + "/metadata-store-realms/" + realm;
+    HttpDelete deleteRequest = new HttpDelete(url);
+    try {
+      HttpResponse response = _forwardHttpClient.execute(deleteRequest);
+      if (response.getStatusLine().getStatusCode() != 200) {
+        HttpEntity respEntity = response.getEntity();
+        LOG.error("the forwarded request to leader has failed for deleteMetadataStoreRealm: {}",
+            respEntity != null ? EntityUtils.toString(respEntity) : "");
```

Comment:
Why is the ternary necessary?

Did you meant to have a log msg like:

"the forwarded request to leader has failed for deleteMetadataStoreRealm:  "?

Reply:
Ditto my reply above.

Reply:
See above